src/strat_engine/blockdev.rs

```diff
@@ -55,87 +55,109 @@ impl BlockDev {
                mda_size: Sectors,
                force: bool)
                -> EngineResult<BlockDev> {
-        let metadata = try!(fs::metadata(path));
+        Ok(try!(Self::batch_new(stratis_id, &[path], mda_size, force)).pop().unwrap())
+    }
```

Comment:
As I pointed out below...it would be great if this function above could be eliminated completely once the engine refactoring is completed.